process/miniblocks/miniblocksProcessor.go

```diff
@@ -89,38 +89,59 @@ func (mp *miniblocksProcessor) prepareMiniblockForDB(
 		Timestamp:       time.Duration(header.GetTimeStamp()),
 	}
 
-	processingType := mp.computeProcessingType(mbIndex, header)
-
 	encodedHeaderHash := hex.EncodeToString(headerHash)
-	if dbMiniblock.SenderShardID == header.GetShardID() {
-		dbMiniblock.SenderBlockHash = encodedHeaderHash
-		dbMiniblock.ProcessingTypeOnSource = processingType
-	} else {
-		dbMiniblock.ReceiverBlockHash = encodedHeaderHash
-		dbMiniblock.ProcessingTypeOnDestination = processingType
-	}
+	isIntraShard := dbMiniblock.SenderShardID == dbMiniblock.ReceiverShardID
+	isCrossOnSource := !isIntraShard && dbMiniblock.SenderShardID == header.GetShardID()
+	if isIntraShard || isCrossOnSource {
+		mp.setFieldsMBIntraShardAndCrossFromMe(mbIndex, header, encodedHeaderHash, dbMiniblock)
 
-	if dbMiniblock.SenderShardID == dbMiniblock.ReceiverShardID {
-		dbMiniblock.ReceiverBlockHash = encodedHeaderHash
-		dbMiniblock.ProcessingTypeOnDestination = processingType
+		return dbMiniblock, nil
 	}
 
+	processingType, _ := mp.computeProcessingTypeAndConstructionState(mbIndex, header)
+	dbMiniblock.ProcessingTypeOnDestination = processingType
+	dbMiniblock.ReceiverBlockHash = encodedHeaderHash
+
 	return dbMiniblock, nil
 }
 
-func (mp *miniblocksProcessor) computeProcessingType(mbIndex int, header coreData.HeaderHandler) string {
+func (mp *miniblocksProcessor) setFieldsMBIntraShardAndCrossFromMe(
+	mbIndex int,
+	header coreData.HeaderHandler,
+	headerHash string,
+	dbMiniblock *data.Miniblock,
+) {
+	processingType, constructionState := mp.computeProcessingTypeAndConstructionState(mbIndex, header)
+
+	dbMiniblock.ProcessingTypeOnSource = processingType
+	switch {
+	case constructionState == int32(block.Final) && processingType == block.Normal.String():
+		dbMiniblock.SenderBlockHash = headerHash
+		dbMiniblock.ReceiverBlockHash = headerHash
```

Comment:
The receiverBlockHash will be overwritten later when the mini block will be executed on destination? If the receiver shard is not self shard then the receiver block hash is not known here.

Reply:
Exactly `ReceiverBlockHash` will be rewritten of the destination shard.

Reply:
ok